plots/plot_utils.cc

```diff
@@ -0,0 +1,119 @@
+#include "plot_utils.h"
+
+std::tuple<std::vector<std::vector<double>>, std::vector<std::vector<double>>,
+           std::vector<std::vector<double>>>
+to_mesh(const Eigen::MatrixXd &grid, const Eigen::VectorXd vals) {
+  // infer the number of points in the ygrid
+  int ny = 0;
+  double first_x = grid(0, 0);
+  while (grid(ny + 1, 0) == first_x) {
```

Comment:
This check screams segmentation fault...

Reply:
That would be a very unusual grid (i.e. evaluate a 2d density over a segment with the same values of the x's)